### PR TITLE
Integrate Lint-Staged w/ Prettier & light ESLint config & Minimal Linting of Current Codebase

### PR DESCRIPTION
**Pull request by CruzMolina**
Ran the entire codebase through eslint using a very minimal setting (no-unused-vars, semi, indent (2 spaces), & block-spacing). Inspired by a quest to eliminate our unused global requires.

There are still a number of unused local vars (~100?) I erred on the side of leaving alone for now, I also only lightly touched truffle-debugger.

Moving forward, I've adding `lint-staged` to run w/ `prettier` and `eslint` (minimal settings) precommit. Hopefully this helps minimize stylistic reviews and bikeshedding.

### CODE REVIEW

File: packages/truffle-debugger/lib/data/actions/index.js [hunk 1]
```diff
@@ -2,30 +2,35 @@ export const SCOPE = "SCOPE";
 export function scope(id, pointer, parentId, sourceId) {
   return {
     type: SCOPE,
-    id, pointer, parentId, sourceId
-  }
+    id,
+pointer,
+parentId,
+sourceId
```
Comment (gnidan) [inline packages/truffle-debugger/lib/data/actions/index.js]: These should probably get indented

File: packages/truffle-debugger/lib/data/actions/index.js [hunk 2]
```diff
@@ -2,30 +2,35 @@ export const SCOPE = "SCOPE";
...
+  };
 }
 
 export const ASSIGN = "ASSIGN";
 export function assign(context, assignments) {
   return {
     type: ASSIGN,
-    context, assignments
+    context,
+assignments
```
Comment (gnidan) [inline packages/truffle-debugger/lib/data/actions/index.js]: (indent)

File: packages/truffle-debugger/lib/data/actions/index.js [hunk 3]
```diff
@@ -2,30 +2,35 @@ export const SCOPE = "SCOPE";
...
   };
 }
 
 export const MAP_KEY = "MAP_KEY";
 export function mapKey(id, key) {
   return {
     type: MAP_KEY,
-    id, key
+    id,
+key
```
Comment (gnidan) [inline packages/truffle-debugger/lib/data/actions/index.js]: (indent)

**Comment by CruzMolina** [comment]
@gnidan I can also push an updated package.json with the eslint devDeps and a pretest script to include in CI. Might end up being noise instead of helpful though since there are ~250 warnings that pop up.

**Pull request merged**